logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/PendingEventsBuffer.kt

```diff
@@ -0,0 +1,59 @@
+package com.wire.kalium.logic.sync
+
+import com.wire.kalium.logic.data.event.Event
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * Stores pending events as they are collected, used to
+ * check if an event was already collected (duplicated, present in both pending and live sources).
+ *
+ * All operations are thread-safe.
+ */
+internal class PendingEventsBuffer {
+    private val events = mutableListOf<Event>()
+    private val mutex = Mutex()
+
+    /**
+     * Adds an [event] to the end of this storage.
+     */
+    suspend fun addToBuffer(event: Event) = mutex.withLock {
```

Comment:
it's a little picky! but isn't better to user add, remove, duplicate and... for methods exist in the class?

Reply:
It's alright. This is fruit of an oversight of mine.

These methods were moved from inside `SyncManager` on another, and made into a class of its own. So I forgot to remove the `toBuffer` suffix.

I'll fix it.